python/utils/bootstrap_utilities.py

```diff
@@ -239,31 +249,44 @@ def start_app(engine):
         return 0
 
 
-def handle_error(data):
+def handle_error(data, proxy=None):
     """
     Attempt to communicate the error back to the GUI proxy given a data
     dictionary like the one passed to the launch_python hook. Note that
     if the server has already been closed (process crashes or user left the
     project) this will actually fail silently, which is alright as the main
     process doesnt't care about this one anymore.
+
+    :param proxy: An optional RPCProxy object to use when sending the
+        error to the server. If a proxy is not given, a client connection
+        will be created on the fly.
     """
+    # build a message for the GUI signaling that an error occurred
+    exc_type, exc_value, exc_traceback = sys.exc_info()
```

Comment:
Maybe check for your flag on the exception here intead, so the fix is self-contained to this file.

Reply:
Cool, sounds very reasonable!